ext/io/lib.rs

```diff
@@ -736,19 +767,97 @@ impl crate::fs::File for StdFileResourceInner {
     self: Rc<Self>,
     mut buf: BufMutView,
   ) -> FsResult<(usize, BufMutView)> {
-    self
-      .with_inner_blocking_task(|file| {
-        let nread = file.read(&mut buf)?;
-        Ok((nread, buf))
-      })
-      .await
+    match &self.kind {
+      /* On Windows, we need to handle special read cancellation logic for stdin */
+      #[cfg(windows)]
+      StdFileResourceKind::Stdin(state) => {
+        loop {
+          let state = state.clone();
+
+          let fut = self.with_inner_blocking_task(move |file| {
+            /* Start reading, and set the reading flag to true */
+            state.lock().unwrap().reading = true;
```

Comment:
Is a flag sufficient? Could multiple threads be in here at once?

Reply:
Yeah, there may be multiple threads in here at once. The reading flag tells tty mode changes to issue read cancellation before proceeding